packages/babel-plugin/src/utils/evaluate-expression.tsx

```diff
@@ -205,6 +206,90 @@ const traverseFunction = (expression: t.Function, meta: Metadata) => {
   return createResultPair(value as t.Expression, updatedMeta);
 };
 
+/**
+ * Will find the function node for the call expression and wrap an IIFE around it (to avoid name collision)
+ * and move all the parameters mapped to passed arguments in the IIFE's scope (own scope in this case).
+ * It will also set own scope path so that when we recursively evaluate any node,
+ * we will look for its binding in own scope first, then parent scope.
+ *
+ * @param expression Expression we want to interrogate.
+ * @param meta Meta data used during the transformation.
+ */
+const traverseCallExpression = (expression: t.CallExpression, meta: Metadata) => {
+  const callee = expression.callee;
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  if (t.isExpression(callee)) {
+    let functionNode = null;
+
+    if (t.isIdentifier(callee)) {
```

Comment:
nit: could we simplify the start by pumping everything through `evaluateExpression`?

Reply:
Oh..we cant actually because `evaluateExpression` will evaluate things and we just need the `functionNode` reference.
For eg. 

```ts
function abcd(color) {
  return {
    color
  }
}

abcd('blue');
```

will return `ObjectExpression` `{ color }` if we use `evaluateExpression`, but we only wanted that function node so that we can isolate the parameters and map them to args.

Reply:
ah ok